### PR TITLE
Add clone options for creating baseline

### PR DESCRIPTION
**Pull request by johnsonm325**

### CODE REVIEW

**Comment by beav** [comment]
code looks OK to me, given that it's stub code to show something in the UI for a demo.

will test in 5-10 min

File: src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js [hunk 1]
```diff
@@ -14,12 +14,27 @@ class CreateBaselineModal extends Component {
         this.submitBaselineName = this.submitBaselineName.bind(this);
 
         this.state = {
-            baselineName: ''
+            baselineName: '',
+            check1: false,
+            check2: false,
+            check3: false
```
Comment (ehelms) [inline src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js]: I know this is a kinda POC, but if its also the starter work for this change, I would ask we consider giving these checks names that are meaningful.

**Comment by beav** [comment]
can you squash this into one commit?

File: src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js [hunk 2]
```diff
@@ -41,10 +70,67 @@ class CreateBaselineModal extends Component {
...
-        const { baselineName } = this.state;
+    renderRadioButtons() {
+        const { fromScratchChecked, copyBaselineChecked, copySystemChecked } = this.state;
+
+        return (<React.Fragment>
+            <Radio
+                isChecked={ fromScratchChecked }
+                id='create baseline'
+                label='Create baseline from scratch'
+                name='radio-1'
```
Comment (beav) [inline src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js]: what's the difference between name and id? does name have to be radio-1, radio-2, radio-3?

Comment (johnsonm325) [reply src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js]: I built it like the example from patternfly, but honestly, I think we could trash the name field. What do you think?

```
import React from 'react';
import { Radio } from '@patternfly/react-core';

class ControlledRadio extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      check1: false,
    };

    this.handleChange = (_, event) => {
      const { value } = event.currentTarget;
      this.setState({ [value]: true });
    };
  }

  render() {
    return (
      <React.Fragment>
        <Radio
          isChecked={this.state.check1}
          name="radio-1"
          onChange={this.handleChange}
          label="Controlled radio"
          id="radio-controlled"
          value="check1"
        />
      </React.Fragment>
    );
  }
}
```

**Comment by beav** [comment]
the multi-select needs to be removed from baseline and system select:

image

**Comment by beav** [comment]
also the "copy from existing system" modal only shows one line since it has a search bar. I don't have an issue with that but you may want to run it by katie

File: src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js [hunk 3]
```diff
@@ -14,22 +16,49 @@ class CreateBaselineModal extends Component {
...
 
     async submitBaselineName() {
-        const { baselineName } = this.state;
-        const { createBaseline, toggleCreateBaselineModal } = this.props;
+        const { baselineName, fromScratchChecked } = this.state;
+        const { createBaseline, toggleCreateBaselineModal, selectedBaselineIds, entities } = this.props;
+        let newBaselineObject;
+
         /*eslint-disable camelcase*/
-        let newBaselineObject = { display_name: baselineName, baseline_facts: []};
```
Comment (ehelms) [inline src/SmartComponents/BaselinesPage/CreateBaselineModal/CreateBaselineModal.js]: If you kept this but dropped the last param then the if-else would be a bit cleaner I think:

```
        if (fromScratchChecked) {
            newBaselineObject['baseline_facts'] =  [];
            await createBaseline(newBaselineObject);
        } else if (selectedBaselineIds.length === 1) {
            await createBaseline(newBaselineObject, selectedBaselineIds[0]);
        } else if (entities.selectedSystemIds.length === 1) {
            newBaselineObject['inventory_uuid'] = entities.selectedSystemIds[0];
            await createBaseline(newBaselineObject);
        }
```

File: src/SmartComponents/BaselinesTable/BaselinesTable.js [hunk 4]
```diff
@@ -18,7 +18,10 @@ import { baselinesTableActions } from './redux';
 class BaselinesTable extends Component {
     constructor(props) {
         super(props);
-        this.onSelect = this.onSelect.bind(this);
+
+        this.state = {
+            createBaselineModal: this.props.createBaselineModal
+        };
```
Comment (ehelms) [inline src/SmartComponents/BaselinesTable/BaselinesTable.js]: Is there an advatange to doing this here for in `mapStateToProps` ?

File: src/SmartComponents/BaselinesTable/BaselinesTable.js [hunk 5]
```diff
@@ -91,22 +107,31 @@ class BaselinesTable extends Component {
         return table;
     }
 
+    createModalRows() {
```
Comment (ehelms) [inline src/SmartComponents/BaselinesTable/BaselinesTable.js]: What makes a modal row a modal row vs. in the normal table?

File: src/SmartComponents/BaselinesTable/redux/baselinesTableReducer.js [hunk 6]
```diff
@@ -61,6 +61,33 @@ export function baselinesTableReducer(state = initialState, action) {
...
+
+            state.baselineTableData.map(row => {
+                if (action.payload.id.includes(row[0])) {
+                    row.selected = action.payload.isSelected;
+                } else {
+                    row.selected = false;
+                }
+
+                newBaselineTableData.push(row);
+            });
```
Comment (ehelms) [inline src/SmartComponents/BaselinesTable/redux/baselinesTableReducer.js]: I think if we re-factored the logic in the SELECT_BASELINE reducer to some helpers, we could use one action to handle both cases. Not required for this change and I might take a crack at doing that.

File: src/SmartComponents/BaselinesTable/BaselinesTable.js (hunk 5 above)
Comment (johnsonm325) [reply src/SmartComponents/BaselinesTable/BaselinesTable.js]: At the moment, modalRows have the checkboxes and don't have the kebab. The normal table has kebab but no checkboxes.

File: src/SmartComponents/BaselinesTable/redux/baselinesTableReducer.js (hunk 6 above)
Comment (johnsonm325) [reply src/SmartComponents/BaselinesTable/redux/baselinesTableReducer.js]: Agree, but definitely want to tackle this later. I'll make a card for it so we ensure it gets addressed.

File: src/SmartComponents/BaselinesTable/BaselinesTable.js (hunk 5 above)
Comment (johnsonm325) [reply src/SmartComponents/BaselinesTable/BaselinesTable.js]: modalRows have checkboxes and no kebabs

**Comment by beav** [comment]
@ehelms any further comment?

**Pull request merged**